cmd/clusterctl/client/repository/template.go

```diff
@@ -42,13 +42,21 @@ type Template interface {
 
 	// Objs returns the cluster template as a list of Unstructured objects.
 	Objs() []unstructured.Unstructured
+
+	//DefaultValues return all varialbe and default commands
+	DefaultValues() map[string]string
+
+	//RequiredFields returns all required Fields
+	RequiredFields() []string
```

Comment:
Wondering if we should instead change `Variables []string` into `Variables []VarablesMetadata` with
```
type VarablesMetadata struct (or interface) {
   Name string
   Required bool
   DefaultValue string (or interface{})
}
```

@vincepri @CecileRobertMichon opinions?
This IMO makes the model more future proof, but the change is a breaking change which is fine for v1alpha4, a little bit more problematic for v1alpha3 even thought there are con API compatibility guarantees for clusterctl

Reply:
the reason I didn't change datatype was that it was being used by other class as []string

Reply:
How about adding a new field `VariablesWithDefaults []VariablesMetadata` and keeping `Variables` for backwards compatibility but marking it as deprecated?

Reply:
I'm ok with either approach, a breaking change sounds fine to me for v0.4.0

Reply:
Ok, then let's go for changing `Variables []string` into `Variables []VariablesMetadata` for v1alpha4, we will reconsider `VariablesWithDefaults []VariablesMetadata` for backporting in required (TBH I think we should avoid to backport if possible).

Reply:
@mabikash Are you able to make the change described in these comments?

Reply:
yes it is fixed on latest code